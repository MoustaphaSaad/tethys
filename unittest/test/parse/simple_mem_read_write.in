proc main
	i32.load r0 1
	i32.load r1 2

	i32.sub sp 4 ; allocate i32
	i32.write sp r0

	i32.sub sp 4 ; allocate i32
	i32.write sp r1

	i32.load r0 3
	i32.load r1 4

	i32.read r1 sp
	i32.add sp 4 ; free i32

	i32.read r0 sp
	i32.add sp 4 ; free i32

	halt
end