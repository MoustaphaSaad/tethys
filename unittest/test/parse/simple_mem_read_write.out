PROC main
  i32.load r0 1
  i32.load r1 2
  i32.load r2 4
  i32.write sp r0
  i32.sub sp r2
  i32.write sp r1
  i32.sub sp r2
  i32.load r0 3
  i32.load r1 4
  i32.add sp r2
  i32.read r1 sp
  i32.add sp r2
  i32.read r0 sp
  halt
END
